src/browser/fetch/request.rs

```diff
@@ -99,6 +100,13 @@ impl<'a> Request<'a> {
         self.header(Header::content_type("text/plain; charset=utf-8"))
     }
 
+    /// Set request body to the provided bytes.
+    /// It will also set `Content-Type` header to `application/octet-stream`.
+    pub fn bytes(mut self, bytes: impl AsRef<[u8]>) -> Self {
+        self.body = Some(Uint8Array::from(bytes.as_ref()).into());
```

Comment:
As far as I can tell both those examples convert from existing `JsValue`s, not basic rust bytes. Is the Safari issue unique to `Uint8Array`? I could try converting to something else before using `Uint8Array::new`.

Reply:
See https://github.com/seed-rs/seed/issues/427 for more info. I don't know how much it's related to the new code - please look at it and choose the best option.
I don't have Safari/Mac to test it and Safari in CI is broken (if I'm not wrong).

Reply:
I also don't have a Mac.

The particular `From` `impl` used by slices for `Uint8Array::new` is

```rust
impl<'a> From<&'a [$ty]> for $name {
    #[inline]
    fn from(slice: &'a [$ty]) -> $name {
        // This is safe because the `new` function makes a copy if its argument is a TypedArray
        unsafe { $name::new(&$name::view(slice)) }
    }
}
```
```rust
/// Creates a JS typed array which is a view into wasm's linear
/// memory at the specified pointer with specified length.
///
/// This function returns a new typed array which is a view into
/// wasm's memory. This view does not copy the underlying data.
///
/// # Unsafety
///
/// Views into WebAssembly memory are only valid so long as the
/// backing buffer isn't resized in JS. Once this function is called
/// any future calls to `Box::new` (or malloc of any form) may cause
/// the returned value here to be invalidated. Use with caution!
///
/// Additionally the returned object can be safely mutated,
/// the changes are guranteed to be reflected in the input array.
pub unsafe fn view_mut_raw(ptr: *mut $ty, length: usize) -> $name {
    let buf = wasm_bindgen::memory();
    let mem = buf.unchecked_ref::<WebAssembly::Memory>();
    $name::new_with_byte_offset_and_length(
        &mem.buffer(),
        ptr as u32,
        length as u32
    )
}
```

As far as I can tell it does not look like it would have the same issues that converting form an `ArrayBuffer` did but I can't find that particular `impl`. Maybe they removed it...

Reply:
Looking deeper I noticed that #427 was converting from a `JsValue` not `ArrayBuffer` but my point about them having probably very different `From` `impl`s should still stand. I can't see the `impl` for `JsValue` though because the js_sys macros make things a little hard to navigate.

Reply:
Could you add a simple unit test to make sure it works? Something like https://github.com/seed-rs/seed/blob/867c679c6729a5c7f545a42720865d55fb1b1883/src/browser/fetch/response.rs#L124-L134. Then we can merge it.

Reply:
Done

Reply:
Done